transport/src/main/java/io/netty/channel/ChannelFactoriesRegistry.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This class provides a default mapping between {@link EventLoopGroup} implementations and {@link ChannelFactory}
+ * implementations
+ */
+public abstract class ChannelFactoriesRegistry {
+
+    private static final class ChannelFactoryDefaults {
+        private final Class<? extends EventLoopGroup> eventLoopGroupType;
+        private final ChannelFactory clientSideFactory;
+        private final ChannelFactory serverSideFactory;
+
+        public ChannelFactoryDefaults(Class<? extends EventLoopGroup> eventLoopGroupType,
+            ChannelFactory clientSideFactory, ChannelFactory serverSideFactory) {
```

Comment:
nit: I believe this needs an extra indent and I believe (unfortunately) is to actually align the args. Please double check and/or somebody can confirm.